src/construction/casting.rs

```diff
@@ -313,27 +299,23 @@ impl ApInt {
     ///
     /// This operation will forward to
     ///
-    /// - [`truncate`](struct.ApInt.html#method.truncate)
-    ///   if `target_width` is less than or equal to the width of
-    ///   the given `ApInt`
-    /// - [`sign_extend`](struct.ApInt.html#method.sign_extend)
-    ///   otherwise
+    /// - [`truncate`](struct.ApInt.html#method.truncate) if `target_width` is less than or equal to
```

Comment:
I have the feeling that this PR is only soooo immensely big because you merged reorganization with auto formatting. Please make PRs with a singular step in the future. So this one huge PR would be at least two PRs.